fs/fcb/src/fcb_bmark.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "os/mynewt.h"
+
+#if MYNEWT_VAL(LOG_FCB_BOOKMARKS)
+
+#include "fcb/fcb.h"
+#include "fcb_priv.h"
+#include "string.h"
+
+void
+fcb_log_init_bmarks(struct fcb_log *fcb_log,
+                    struct fcb_log_bmark *buf, int bmark_count)
+{
+    fcb_log->fl_bset = (struct fcb_log_bset) {
+        .fls_bmarks = buf,
+        .fls_cap = bmark_count,
```

Comment:
I am making this comment just so I am sure I understand this, but would you ever expect a buffer to not be sized based on bmark_count? I do not think this is a big deal but I was wondering if the code should check to see if buf is sized appropriately (based on the bmark_count). I guess if it did that it could calculate the # of bookmarks. No need to change anything here btw.

Reply:
I do think that is a legitimate danger.  The size of `buf` may not match `bmark_count`.  I don't think there is a good way around this, though.  This is just a limitation of C - you can't pass an array to a function, you can only pass a pointer.  So it is always the user's responsibility to specify the correct size as a separate argument.